packages/tenant-process/src/services/tenantService.ts

```diff
@@ -451,6 +456,106 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async verifyVerifiedAttribute({
+      tenantId,
+      tenantAttributeSeed,
+      organizationId,
+      correlationId,
+    }: {
+      tenantId: TenantId;
+      tenantAttributeSeed: ApiVerifiedTenantAttributeSeed;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<Tenant> {
+      logger.info(
+        `Verifying attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+
+      if (organizationId === tenantId) {
+        throw verifiedAttributeSelfVerification();
+      }
+
+      await assertAttributeVerificationAllowed({
+        producerId: organizationId,
+        consumerId: tenantId,
+        attributeId: unsafeBrandId(tenantAttributeSeed.id),
+        readModelService,
+      });
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const verifiedTenantAttribute = targetTenant.data.attributes.find(
+        (attr) =>
+          attr.type === tenantAttributeType.VERIFIED &&
+          attr.id === tenantAttributeSeed.id
+      ) as VerifiedTenantAttribute;
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...targetTenant.data,
+        updatedAt: new Date(),
+      };
+      if (!verifiedTenantAttribute) {
+        // assigning attribute for the first time
+        updatedTenant = {
+          ...updatedTenant,
+          attributes: [
+            {
+              id: unsafeBrandId(tenantAttributeSeed.id),
+              type: tenantAttributeType.VERIFIED,
+              assignmentTimestamp: new Date(),
+              verifiedBy: [
+                {
+                  id: organizationId,
+                  verificationDate: new Date(),
+                  expirationDate: tenantAttributeSeed.expirationDate
+                    ? new Date(tenantAttributeSeed.expirationDate)
+                    : undefined,
+                  extensionDate: tenantAttributeSeed.expirationDate
+                    ? new Date(tenantAttributeSeed.expirationDate)
+                    : undefined,
+                },
+              ],
+              revokedBy: [],
+            },
+          ],
+        };
+      } else {
```

Comment:
Same as the point made in #381, IMHO there should be a check for the verified attribute existence in the read model

Reply:
I agree, this check should be added